nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp

```diff
@@ -49,32 +52,66 @@ GoalUpdater::GoalUpdater(
     rclcpp::SystemDefaultsQoS(),
     std::bind(&GoalUpdater::callback_updated_goal, this, _1),
     sub_option);
+  goals_sub_ = node_->create_subscription<nav2_msgs::msg::PoseStampedArray>(
+    goals_updater_topic,
+    rclcpp::SystemDefaultsQoS(),
+    std::bind(&GoalUpdater::callback_updated_goals, this, _1),
+    sub_option);
 }
 
 inline BT::NodeStatus GoalUpdater::tick()
 {
   geometry_msgs::msg::PoseStamped goal;
+  Goals goals;
 
   getInput("input_goal", goal);
+  getInput("input_goals", goals);
 
   // Spin multiple times due to rclcpp regression in Jazzy requiring a 'warm up' spin
   callback_group_executor_.spin_all(std::chrono::milliseconds(1));
   callback_group_executor_.spin_all(std::chrono::milliseconds(49));
 
-  if (last_goal_received_.header.stamp != rclcpp::Time(0)) {
+  if (last_goal_received_.header.stamp == rclcpp::Time(0)) {
+    // if the goal doesn't have a timestamp, we don't do any timestamp-checking and accept it
+    setOutput("output_goal", last_goal_received_);
+  } else {
     auto last_goal_received_time = rclcpp::Time(last_goal_received_.header.stamp);
     auto goal_time = rclcpp::Time(goal.header.stamp);
     if (last_goal_received_time > goal_time) {
-      goal = last_goal_received_;
+      setOutput("output_goal", last_goal_received_);
     } else {
       RCLCPP_WARN(
         node_->get_logger(), "The timestamp of the received goal (%f) is older than the "
         "current goal (%f). Ignoring the received goal.",
         last_goal_received_time.seconds(), goal_time.seconds());
+      setOutput("output_goal", goal);
+    }
+  }
+
+  if (last_goals_received_.poses.empty()) {
+    setOutput("output_goals", goals);
+  } else if (last_goals_received_.header.stamp == rclcpp::Time(0)) {
+    setOutput("output_goals", last_goals_received_.poses);
```

Comment:
So this would then return `goals` instead with an error log. Maybe also just reset the last goals received to an empty message so it only logs once

Reply:
(assuming we agree that a `rclcpp::Time(0)` should be ignored)